### PR TITLE
kernel/binary_manager: Support reloading optimization for common binary

### PR DESCRIPTION
**Pull request by jeongarmy**
kernel/binary_manager: Support reloading optimization for common binary

### CODE REVIEW

File: os/kernel/binary_manager/binary_manager_recovery.c [hunk 1]
```diff
@@ -350,9 +353,9 @@ void binary_manager_recovery(int bin_idx)
 		bmlldbg("Failed to deactivate binary, bin idx %d\n", bin_idx);
 		goto reboot_board;
 	}
+	BIN_STATE(bin_idx) = BINARY_FAULT;
```
Comment (jeongchanKim) [inline os/kernel/binary_manager/binary_manager_recovery.c:356]: How about moving this into the `binary_manager_deactivate_binary`?

Comment (jeongarmy) [reply os/kernel/binary_manager/binary_manager_recovery.c]: It's good. I modified.

File: os/kernel/binary_manager/binary_manager_recovery.c [hunk 2]
```diff
@@ -334,14 +337,16 @@ void binary_manager_recovery(int bin_idx)
 	/* If a fault happens in common or user binaries, we need to reload the library and all user binaries */
 	int bidx;
 	int bin_count = binary_manager_get_ucount();
+
+	BIN_STATE(BM_CMNLIB_IDX) = BINARY_FAULT;
```
Comment (sunghan-chang) [inline os/kernel/binary_manager/binary_manager_recovery.c:341]: Please leave a comment why you change the state of common lib unlike others.

Comment (sunghan-chang) [reply os/kernel/binary_manager/binary_manager_recovery.c:347]: As we discussed on the chat, let's move this into deactivate function like other and do changing the state for common there. Then, it's much better to understand and it's better to maintain.

File: os/kernel/binary_manager/binary_manager_load.c [hunk 3]
```diff
@@ -122,14 +125,10 @@ static int binary_manager_load_binary(int bin_idx, char *path, load_attr_t *load
...
-			if (bin_idx == 0) {
-				printf("%s Load 'common'(%s) success!\n", BINARY_COMP_TYPE, path);
-			} else {
-				printf("%s Load '%s'(%s) success! pid = %d\n", BINARY_COMP_TYPE, load_attr->bin_name, path, ret);
-			}
 			/* Set the data in table from header */
 			BIN_LOAD_ATTR(bin_idx) = *load_attr;
 			strncpy(BIN_NAME(bin_idx), load_attr->bin_name, BIN_NAME_MAX);
+			printf("Load success! [Name: %s] [Version: %d] [Partition: %s] %s\n", BIN_NAME(bin_idx), BIN_LOADVER(bin_idx), GET_PARTNAME(BIN_USEIDX(bin_idx)), BINARY_COMP_TYPE);
 			return OK;
```
Comment (jeongchanKim) [inline os/kernel/binary_manager/binary_manager_load.c:132]: How about returning `BINMGR_OK`?

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**